### ISSUE TITLE
Trying to use MK2 Swerve Modules

### ISSUE THREAD

**Issue by willemmy** [SwerveDriveSpecialties/swerve-template#6]
2022 Season Swerve-template is building fine, but crashing repeatedly on deploy.

Using Neo motors and CanSparkMax controllers. Using the Absolute hall encoder.

Can anyone help us out as to why the crashing? We cannot seem to find anywhere in the library where there could be a problem.

Any help would be great!

TIA!

**Comment by jasondaming** [comment]
The swerve template uses the swerve-lib that has not been updated for 2022.  Use my [repo](https://github.com/jasondaming/2022-6391-Swerve/tree/master) until they release an update.

**Comment by willemmy** [comment]
OK Will give that a try. Thank you Jason!!!

**Comment by willemmy** [comment]
Jason, 

Looking thru the library in the MK3 builders I only see CanCoder ID's. Is there a lib that will use the MA3-A10-250-N encoder like we had in 2019 season? We opened our code from 2019 swerve and it converted to the 2022 season and would not build. This has led us to where we are today.....At this point we do not have CTRE CanCoders.

**Comment by jasondaming** [comment]
SDS didn't support the MK2 in their library.  I am not sure what their plans are for the upcoming release.  All that has been said is the MK4i.

What code were you using in 2019?  SDS [Example-Swerve](https://github.com/SwerveDriveSpecialties/Example-Swerve) seems to be more catered towards MK2.

You had the MA3 going back to the RIO?

**Comment by willemmy** [comment]
Yes in 2019 we download the Example-Swerve and altered the CanSparkMax id's
changed the gyro to a navx
Altered the encoder port numbers (AnalogInputs) and the MA3's were wired into the Rio

Deployed the code and set the 4 Offset variables, re-deployed and it ran great!
Only got to use it in one competition before the covid shutdown.
At this point after updating to the 2022 season we have no way to drive the robot.
